packages/@headlessui-react/src/components/combobox/combobox.tsx

```diff
@@ -1753,25 +1767,21 @@ function OptionFn<
 
   let enableScrollIntoView = useRef(data.virtual || data.__demoMode ? false : true)
   useIsoMorphicEffect(() => {
-    if (!data.virtual) return
-    if (!data.__demoMode) return
```

Comment:
These conditions were incorrect. We _don't_ want to scroll into view when virtual mode is used or when demo mode is used.